File: src/com/bridgelabz/functional/FlipCoin.java
/*
 * @purpose	:Flip a coin and calculate percentage of head and tail
 * @author	:sangita awaghad
 * @since	:29-07-2019
 */
package com.bridgelabz.functional;
import java.util.Scanner;

import com.bridgelabz.utility.Utility;

public class FlipCoin {

	public static void main(String args[])
	{
		Utility utility=new Utility();
		Scanner scanner=new Scanner(System.in);
		int mNumberOfTime,mTrails=0,mHeadCount=0,mTailCount=0;
		
		
		System.out.println("How Many time You want to flip coin");
		mNumberOfTime=scanner.nextInt();

		while(mNumberOfTime>mTrails)
		{
			if(utility.giveRandomNumber()>0.5)
			{
				System.out.println("You Flip Tail");
				mHeadCount++;
			}
			else
			{
				System.out.println("You Flip Head");
				mTailCount++;
			}
			mTrails++;
		}
		System.out.println("Number of times Head Flip: "+mHeadCount);
		System.out.println("Number of times Tail Flip: "+mTailCount);
		
		System.out.println("Head Percentage: "+utility.calculatePercentage(mNumberOfTime, mHeadCount));
		System.out.println("Tail Percentage: "+(100f-utility.calculatePercentage(mNumberOfTime, mHeadCount)));
	}
}

File: nodeProject/FlipCoin.js
const utility=require('./Utility/Utility');
console.log("How many times you want to filp");
var noOfFilips=utility.getInputNumber();
var headCount=0;
var tailCount=0;
do{
    var rand=Math.random();
    if(rand<0.5)
    {
        headCount++;
        console.log("you flip HEAD");
    }
    else
    {
        tailCount++;
        console.log("you flip Tail");
    }
}while((headCount+tailCount)<noOfFilips);
console.log("HEAD Percentage :"+headCount/noOfFilips*100);
console.log("TAIL Percentage :"+tailCount/noOfFilips*100);
